router/auth.js

```diff
@@ -5,17 +5,19 @@ const router = express.Router()
 
 router.post('/register',async(req,res)=>{
 
-   const{name,email,phoneno,password}=req.body;
-   if(!name||!email||!phoneno||!password){
+   const{name,email,createpassword,confirmpassword}=req.body;
+   if(!name||!email||!createpassword||!confirmpassword){
       res.status(601).json({error:"fill all the forms correctly"}) 
    } 
    
    try {
-    const data= await mdata.findOne({phoneno:phoneno})
+    const data= await mdata.findOne({email:email})
 
     if(data){
-    return res.status(401).json({error:"user already registered"}) 
-    }
+      console.log("user registered already") 
+   return res.status(401).json({error:"user already registered"})  
```

Comment:
Status 401 indicates unauthenticated meaning that the request needs authentication, but in this case, this route is for users to register and so authentication is not needed. Instead we should return status 400 to indicate the client has sent wrongly/unavailable information.
For more information about HTTP response status codes visit: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
```suggestion
   return res.status(400).json({error:"user already registered"})  
```